### PR TITLE
Add support to wrap prover for verifying chunked step proofs

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR fixes the remaining bugs in the wrap prover for handling chunked step proofs.

To achieve this, this PR
* integrates https://github.com/o1-labs/proof-systems/pull/1279, to remove the hard-coded assumption that `zk_rows = 3` in the generated expression
* updates the types to marshall around the chunked public input evaluations
* adds a `num_chunks` parameter to `Pickles.compile`, to allow for the creation of proofs using a different number of chunks.

With these changes, this PR now passes the local test for creating proofs using chunks in pickles. The last remaining stage is verifying those proofs recursively using the step circuit, which will be completed in an upcoming PR.

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/pickles/pickles.ml [hunk 1]
```diff
@@ -1561,6 +1563,7 @@ module Make_str (_ : Wire_types.Concrete) = struct
                           ~endo:Endo.Step_inner_curve.base
                           ~mds:Tick_field_sponge.params.mds
                           ~srs_length_log2:Common.Max_degree.step_log2
+                          ~zk_rows:3
```
Comment (dannywillems) [inline src/lib/pickles/pickles.ml:1566]: Nit: having a constant somewhere we use everywhere would be nice.

**Review by dannywillems — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by mitschabaude — APPROVED** [review]
approving o1js bump

**Pull request merged**